clang/lib/Serialization/TemplateArgumentHasher.cpp

```diff
@@ -65,7 +65,9 @@ void TemplateArgumentHasher::AddTemplateArgument(TemplateArgument TA) {
 
   switch (Kind) {
   case TemplateArgument::Null:
-    llvm_unreachable("Expected valid TemplateArgument");
+    // These can occur in incomplete substitutions performed with code
+    // completion (see PartialOverloading).
```

Comment:
`see PartialOverloading` where? Is there a specific method or API doc we should look at?

Reply:
It's the term of art within clang, and the name of the flag as used in source code.
A search on doxygen for this term should lead to:
```C++
  /// AddOverloadCandidate - Adds the given function to the set of
  /// candidate functions, using the given function call arguments.  If
  /// @p SuppressUserConversions, then don't allow user-defined
  /// conversions via constructors or conversion operators.
  ///
  /// \param PartialOverloading true if we are performing "partial" overloading
  /// based on an incomplete set of function arguments. This feature is used by
  /// code completion.
```

This term can also easily be greped for, and there are no irrelevant results.
But this is mostly undocumented.

Reply:
I think this name is the best stable reference to this functionality.
And I don't think mentioning file names is a good idea, and there is no documentation to point to.

Is there any specific thing this comment should say so you think this would be easier to find?